labs/sysdev-u-boot-stm32mp2/sysdev-u-boot-stm32mp2.tex

```diff
@@ -0,0 +1,590 @@
+\bashcmd{$ export ARCH=arm}
+
+\item Run \inlinebash{$ make <NAME>_defconfig}, where the list of available
+  configurations can be found in the \code{configs/} directory. We will use the standard one
+  (\code{stm32mp25}).
+
+\item Now that you have a valid initial configuration, you can now
+  run \inlinebash{$ make menuconfig} to further edit your bootloader features.
+
+  \begin{itemize}
+
+  \item In the \code{Environment} submenu, we will configure U-Boot so
+    that it stores its environment inside a file called \code{uboot.env}
+    in an ext4 filesystem:
+    \begin{itemize}
+    \item Disable \code{Environment is not stored}. We want changes to variables to
+        be persistent across reboots
+    \item Enable \code{Environment is in a EXT4 filesystem}. Disable all other
+        options for environment storage (e.g. MMC, SPI, UBI)
+% Environment in MMC also works, but we need to be careful with the size available
+% and the offset defined. tests showed that if MMC is selected it will take
+% precedence over ext4
+    \item The value for \code{Name of the block device for the environment} should be \code{mmc}
+    \item The value for \code{Device and partition for where to store
+      the environment in EXT4} should be \code{0:3}, which indicates
+      we want to store the environment in the 3th partition of the
+      first MMC device.
+    \item The value for \code{Name of the EXT4 file to use for the
+      environment} should be \code{/uboot.env}, which indicates the
+      filename inside which the U-Boot environment will be stored
+    \end{itemize}
+
+  \item In the \code{Device Drivers} $\rightarrow$ \code{Watchdog
+      Timer Support} submenu, disable \code{IWDG watchdog
+      driver for STM32 MP's family}, so that U-Boot doesn't start the
+    watchdog.
+  \end{itemize}
+
+Install the following packages which should be needed to compile U-Boot for
+your board:
+
+\begin{bashinput}
+$ sudo apt install libssl-dev device-tree-compiler swig \
+       python3-dev python3-setuptools uuid-dev libgnutls28-dev
+\end{bashinput}
+
+\item Finally, run \bashcmd{make DEVICE_TREE=stm32mp257f-dk}
+  which will build U-Boot
+  \footnote{You can speed up the
+  compiling by using the \code{-jX} option with \code{make}, where X
+  is the number of parallel jobs used for compiling. Twice the
+  number of CPU cores is a good value.}.
+  The \code{DEVICE_TREE} variable specifies the specific
+  Device Tree that describes our hardware board.
+
+{\bf Note}: u-boot build may fail on your machine if you have a recent
+version of python. Such issue is
+\href{https://source.denx.de/u-boot/u-boot/-/commit/a63456b9191fae2fe49f4b121e025792022e3950}{already
+fixed upstream}, but not in the version targeted for the training. To get the relevant fix,
+you can cherry-pick the fix onto your local branch:
+
+\code{git cherry-pick a63456b9191fae2fe49f4b121e025792022e3950}
+\end{enumerate}
+
+\section{op-tee setup}
+
+Get the downstream optee sources:
```

Comment:
"downstream" isn't clear and not used very often in our training. ST-provided, or something like this. Also optee should be OP-TEE.